### PR TITLE
init

### PR DESCRIPTION
**Pull request by MaksymKos**
https://maksymkos.github.io/react_autocomplete/

### CODE REVIEW

File: src/DropDown.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={(event) => {
+            setQuery(event.target.value);
+            applyQuery(event.target.value);
+            setIsDropdownShown(false);
+          }}
```
Comment (VitaliyBondarenko1982) [inline src/DropDown.tsx:68]: move to separate function and pass referense on it here

File: src/DropDown.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  delay,
+  onSelected,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownShown, setIsDropdownShown] = useState(false);
+
+  const applyQuery = useCallback(debounce((newQuery: string) => {
+    setAppliedQuery(newQuery);
+    setIsDropdownShown(true);
```
Comment (VitaliyBondarenko1982) [inline src/DropDown.tsx:32]: show dropdown on onFocus event

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Dropdown shouldn't be visible when input does not in focus
chrome-capture-2023-7-26 (2)

File: src/DropDown.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        { 'is-active': isDropdownShown },
+      )}
+    >
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          onChange={(event) => handleQueryChange(event)}
```
Comment (etojeDenys) [inline src/DropDown.tsx:71]:
```suggestion
          onChange={handleQueryChange}
```
we can simplify it

File: src/DropDown.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      </div>
+
+      <div className="dropdown-menu" role="menu">
+        <div className="dropdown-content">
+          {filteredPeople.length !== 0
+            ? filteredPeople.map(person => (
+              <a
+                href="/"
+                key={person.name}
+                className={`dropdown-item ${person.sex === 'm' ? 'has-text-link' : 'has-text-danger'}`}
```
Comment (etojeDenys) [inline src/DropDown.tsx:84]: let's use className package for such cases

**Review by etojeDenys — CHANGES_REQUESTED** [review]
still can reproduce it
image
feel free to ask in the chat if you have some problems

File: src/DropDown.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        <div className="dropdown-content">
+          {filteredPeople.length !== 0
+            ? filteredPeople.map(person => (
+              <a
+                href="/"
+                key={person.name}
+                className={classNames(
+                  'dropdown-item',
+                  person.sex === 'm' && 'has-text-link',
+                  person.sex === 'f' && 'has-text-danger'
```
Comment (etojeDenys) [inline src/DropDown.tsx:87]:
```suggestion
                   'has-text-link': person.sex === 'm',
                   'has-text-danger': person.sex === 'f'
```

**Review by etojeDenys — CHANGES_REQUESTED** [review]
https://github.com/mate-academy/react_autocomplete/assets/70220667/5b0f97f7-441a-46da-8cf2-add7e75561b7

cannot select the person

**Review by etojeDenys — CHANGES_REQUESTED** [review]
still can reproduce the bug

**Review by etojeDenys — APPROVED** [review]
lgtm

**Pull request closed without merging**